### PR TITLE
feat(kit): `DialogForm` add new service

### LINKED ISSUE #3828: 🚀 - Add helping service to control dialog closing
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

core

### Description

Add helping service to control of dialog closing implemented in https://github.com/Tinkoff/taiga-ui/issues/3181

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3828

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/core/components/dialog/dialog.component.ts [hunk 1]
```diff
@@ -108,3 +99,7 @@ export class TuiDialogComponent<O, I> {
         }
     }
 }
+
+function toObservable<T>(valueOrStream: Observable<T> | T): Observable<T> {
```
Comment (splincode) [inline projects/core/components/dialog/dialog.component.ts:103]: maybe move to `@taiga/cdk/utils`

File: projects/demo-integrations/cypress/tests/core/dialogs/dialogs.cy.ts [hunk 2]
```diff
@@ -75,32 +75,51 @@ describe(`Dialogs`, () => {
...
-
-                cy.get(`tui-prompt .t-button`).last().click();
-
-                cy.tuiShow(`tui-doc-page`);
-
-                cy.tuiWaitBeforeScreenshot().matchImageSnapshot(
-                    `${index}-7-dialogs-prompt-gone`,
-                    {capture: `viewport`},
-                );
+                beforeEach(() => {
```
Comment (splincode) [inline projects/demo-integrations/cypress/tests/core/dialogs/dialogs.cy.ts:78]: image

rename 
```
it(`Dialog with confirmation works`, () => {
``` to 

```
describe(`Dialog with confirmation works`, () => {
```

File: projects/core/components/dialog/dialog.component.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/core/components/dialog/dialog.component.ts:103]: \+ reuse this new util `tuiToObservable` for the similar cases:
Screenshot 2023-03-14 at 10 23 48

File: projects/demo-integrations/cypress/tests/core/dialogs/dialogs.cy.ts [hunk 3]
```diff
@@ -75,32 +75,51 @@ describe(`Dialogs`, () => {
             });
 
             it(`Dialog with confirmation works`, () => {
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/core/dialogs/dialogs.cy.ts:77]:
```suggestion
            describe(`Dialog with confirmation works`, () => {
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/components/dialog/dialog.component.ts (hunk 1 above)
Comment (waterplea) [reply projects/core/components/dialog/dialog.component.ts:103]: i18n does not depend on cdk though

Comment (nsbarsukov) [reply projects/core/components/dialog/dialog.component.ts:103]: > i18n does not depend on cdk though

my mistake)

Comment (waterplea) [reply projects/core/components/dialog/dialog.component.ts:103]: It is only used here, I'll keep it here for now.

**Pull request merged**